### PR TITLE
feat: Also return the interface name

### PR DESCRIPTION
**Pull request by larseggert**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -86,6 +95,12 @@ fn get_interface_mtu_linux_macos(socket: &UdpSocket) -> Result<usize, Error> {
     #[cfg(target_os = "linux")]
     use libc::{ifreq, ioctl};
 
+    fn hash_interface_name(name: &str) -> InterfaceId {
+        let mut hasher = DefaultHasher::new();
+        name.hash(&mut hasher);
+        hasher.finish()
+    }
```
[the remark this answers is not in the document]
Comment (larseggert) [reply src/lib.rs:102]: Yeah, I was looking for a unified interface across OSs. I guess I could make this a `&str` and use `if_indextoname` on Windows?

**Comment by larseggert** [comment]
@djc so how do I go now get updated Windows bindings when I need to add support for a new function? I don't have access to a Windows machine to run bindgen.

**Comment by djc** [comment]
> @djc so how do I go now get updated Windows bindings when I need to add support for a new function? I don't have access to a Windows machine to run bindgen.

I did it via CI, which will print the newly generated bindings when the comparison fails -- but you have to get the integration test to run, which only happens when the library compiles. Might be an option to put the codegen in a separate crate to make that easier...

**Comment by larseggert** [comment]
> I did it via CI, which will print the newly generated bindings when the comparison fails -- but you have to get the integration test to run, which only happens when the library compiles. Might be an option to put the codegen in a separate crate to make that easier...

Hm. This is getting complex fast. I'm tempted to revert #14 and just go back to what we had before...

**Comment by djc** [comment]
If you just make a separate branch/PR that adds the newly required symbols only, that should make things easier.

**Comment by larseggert** [comment]
> but you have to get the integration test to run

Could this happen in `build.rs`? IIRC @martinthomson suggested something similar.

**Comment by djc** [comment]
> > but you have to get the integration test to run
> 
> Could we run bindgen in `build.rs`? IIRC @martinthomson suggested something similar.

Like the previous approach, that optimizes for the maintainers at the [expense](https://github.com/mozilla/mtu/pull/14#discussion_r1741745334) of every downstream build ever, which doesn't seem like a good trade-off to me.

**Comment by larseggert** [comment]
The "separate PR" thing worked, thanks!

**Review by mxinden — APPROVED** [review]
Thanks!

Definitely in favor of returning a UTF-8 String over a hash.

**Pull request merged**